docs/README.md

```diff
@@ -0,0 +1,25 @@
+# Numaflow
```

Comment:
@edlee2121 - we need this file as it's the homepage of numaproj.github.io/numaflow/.

Please suggest the content.